pandas/core/apply.py

```diff
@@ -693,8 +717,8 @@ def values(self):
     def apply(self) -> DataFrame | Series:
         """compute the results"""
         # dispatch to agg
-        if is_list_like(self.func):
-            return self.apply_multiple()
+        if is_list_like(self.func) or is_dict_like(self.func):
```

Comment:
dicts are considered list-like; no need for the 2nd check here.

Reply:
Ok, I changed it. I've changed the comment above instead to explain dicrlike go here too.